.github/workflows/cont_integration.yml

```diff
@@ -18,9 +18,9 @@ jobs:
     strategy:
       matrix:
         rust:
-          - version: 1.65.0 # STABLE
+          - version: stable
```

Comment:
This means that if a new version of rust is released, with a new version of clippy with stricter rules, our ci would immediately fail the clippy check. Are we ok with this?

Reply:
I don't think so because of https://github.com/bitcoindevkit/bdk/blob/52f395555792a2f24ae7248d98da318bdad5dadb/clippy.toml#L1

Reply:
I think it means that clippy won't suggest any improvement that breaks msrv, not that clippy is pinned to version 1.63.0.

Reply:
From the Ref above:

> Projects that intend to support old versions of Rust can disable lints pertaining to newer features by specifying the minimum supported Rust version (MSRV) in the clippy configuration file.

So I think that you worry about

> if a new version of rust is released, with a new version of clippy with stricter rules, our ci would immediately fail the clippy check

would not fail right?

Reply:
I think it would fail.
Let's say that rust 1.80 introduces the method `pizza()`, and clippy 1.80 introduces two new rules:
1. `pizza()` should always be used instead of `format()`
2. Every variable name should start with a `x` (this makes your code more beautiful)

If your msrv is 1.79, clippy wouldn't complain about rule 1: you can't use `pizza()`, as it doesn't exist in rust 1.79! But I'm kind of sure that it would still warn you about rule 2, since it's doable in earlier versions of rust too. 

Anyways, I don't know how often clippy adds rules that don't pertain to new rust features, so it might not be a big deal after all.

Reply:
Yes but first MSRV CI would fail because you cannot include `pizza()` since it does not exist in MSRV.

And even if you don't include it the `clippy.toml` MSRV config will make sure to not enable the `pizza_instead of format` lint because of

> Projects that intend to support old versions of Rust can disable lints pertaining to newer features by specifying the minimum supported Rust version (MSRV) in the clippy configuration file.

Reply:
Rule 2 is a "newer feature" that I've mentioned above.
So CI with `clippy.toml` would pass.